### PR TITLE
707 log slow checkpoints

### PR DESCRIPTION
**Pull request by mitesh-gosavi**
**Description of changes**

The —checkpoint flag has been added to the argument parser for the mplotqueries. The parser, on receiving the —checkpoint flag, searches for loglines containing  the “Checkpoint took”  token (for reference, see https://github.com/mongodb/mongo/commit/f3eb44233050eee4da6aeaf4b4f93e9fb326a4d3). On encountering the token, the parser parses the token after “took” and typecast it to an integer to get the duration followed by mapping the details on to the graph.

**Testing**

The following testing tools have been used to test the newly added code:
1. Tox
2. Flake8
3. iSort
4. Doc8

2. The test result of a code coverage can be seen in table below:

**Name                                              Stmts   Miss  Cover
---------------------------------------------------------------------**

mtools/mplotqueries/mplotqueries.py                 400     93    77%
mtools/util/logevent.py                             510      2    99%

**Version Information**

| macOS 		| 10.14.5	|
| mongod		| v4.0.10	|
| MongoDB shell	| v4.0.10	|
| Python		| 2.7.10		|

### CODE REVIEW

File: mtools/util/logevent.py [hunk 1]
```diff
@@ -202,6 +203,9 @@ def duration(self):
                                      self.line_str)
                 if matchobj:
                     self._duration = int(matchobj.group(1))
+            # SERVER-16176 log the checkpoints
```
Comment (HenryGP) [inline mtools/util/logevent.py]: Logging of **slow** checkpoints

File: mtools/mplotqueries/mplotqueries.py [hunk 2]
```diff
@@ -118,6 +119,11 @@ def __init__(self):
                                     help=("Save the plot to a file instead of "
                                           "displaying it in a window"))
 
+        # SERVER-16176 - checkpoints argument has been added to the existing group
+        self.argparser.add_argument('--checkpoints',
+                                    action='store_true', default=None,
+                                    help=("display the slow transactions "
```
Comment (HenryGP) [inline mtools/mplotqueries/mplotqueries.py]: It is actually slow WiredTiger checkpoints if I am not mistaken. This is not related to actual transactions.

**Review by HenryGP — CHANGES_REQUESTED** [review]
Some minor changes requested from my end :)

File: mtools/test/test_mplotqueries.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class TestMPlotQueries(object):
+
+    def setup(self):
+        """Startup method to create mloginfo tool."""
+        self.tool = MPlotQueriesTool()
+        self._test_init()
+
+    def _test_init(self, filename='mongod_225.log'):
+        # load logfile(s)
+        self.logfile_path = os.path.join(os.path.dirname(mtools.__file__),'test/logfiles/', filename)
```
Comment (thiagogalesi) [inline mtools/test/test_mplotqueries.py:26]: btw, good practice 👍

File: mtools/util/logevent.py [hunk 4]
```diff
@@ -202,6 +203,9 @@ def duration(self):
...
+            elif "Checkpoint" in self.line_str:
+                self._duration = int(line_str[line_str.find("Checkpoint took") + len("Checkpoint took"): -20]) * 1000
```
Comment (thiagogalesi) [inline mtools/util/logevent.py]: See the comment on 706 on how to capture this with a regex

**Review by thiagogalesi — COMMENTED** [review]
Some comments

File: mtools/mplotqueries/mplotqueries.py [hunk 5]
```diff
@@ -19,6 +19,7 @@
     from matplotlib.dates import AutoDateFormatter, date2num, AutoDateLocator
     from matplotlib import __version__ as mpl_version
     import mtools.mplotqueries.plottypes as plottypes
+    import re
```
Comment (manfontan) [inline mtools/mplotqueries/mplotqueries.py]: Review PEP style guide to sort imports https://www.python.org/dev/peps/pep-0008/#imports ;)

File: mtools/test/test_mplotqueries.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import sys
+from datetime import datetime, timedelta
+from random import randrange
+
+from dateutil import parser
+from nose.plugins.skip import SkipTest
+from nose.tools import raises
+
+import os
```
Comment (manfontan) [inline mtools/test/test_mplotqueries.py]: Review PEP style guide to sort imports https://www.python.org/dev/peps/pep-0008/#imports ;)

**Review by HenryGP — APPROVED** [review]

**Review by davidedeangelismdb — APPROVED** [review]

File: mtools/util/logevent.py [hunk 7]
```diff
@@ -202,7 +203,10 @@ def duration(self):
                                      self.line_str)
                 if matchobj:
                     self._duration = int(matchobj.group(1))
-
+            # SERVER-16176 - Logging of slow checkpoints
```
Comment (thiagogalesi) [inline mtools/util/logevent.py:220]: 👍

**Review by thiagogalesi — APPROVED** [review]

**Pull request merged**